sway/tree/view.c

```diff
@@ -715,10 +715,14 @@ void view_map(struct sway_view *view, struct wlr_surface *wlr_surface,
 		input_manager_set_focus(&view->container->node);
 	}
 
-	const char *app_id = view_get_app_id(view);
-	if (app_id != NULL) {
+	const char *app_id;
+	const char *class;
+	if ((app_id = view_get_app_id(view)) != NULL) {
 		wlr_foreign_toplevel_handle_v1_set_app_id(
 				view->foreign_toplevel, app_id);
+	} else if ((class = view_get_class(view)) != NULL) {
+		wlr_foreign_toplevel_handle_v1_set_app_id(
+				view->foreign_toplevel, class);
```

Comment:
What exactly is "instance" in X11 world? I'm interested in keeping stuff in sync across phoc, sway and Wayfire, but I wonder if it's a good idea since multiple instances of Wayland apps are getting the same app-id.

Reply:
https://metacpan.org/pod/X11::Protocol::WM#WM_CLASS
According to this - instance is more closer to app name than class.
But in fact both of them are useful.
For example:
[Class] - [Instance] - [Can find icon by Gtk use ...]
firefox - Navigator - class
Code - code - instance
Skype - skype - instance

Reply:
What if there's a space in the class name, then it's impossible to only get a class with such a change?

Reply:
Any another separate char? (#, $).

Reply:
According to [the spec](https://www.x.org/releases/X11R7.6/doc/xorg-docs/specs/ICCCM/icccm.html#wm_class_property), the class is set by the app and the instance is set by the user (and defaults to `argv[0]`). I'd say this makes instance completely irrelevant here.

Reply:
Additionally, the [desktop entry spec](https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#recognized-keys) has `StartupWMClass` for the class, and nothing for the instance.

Reply:
-> Class names are specified by the application writer.
-> and defaults to argv[0]
argv[0] - program name.

Reply:
A compositor really wanting to do the right thing would build an index of desktop entries and convert the class name to an app_id when necessary. Sway won't.

Reply:
-> The WM_CLASS property (of type STRING without control characters) contains two consecutive null-terminated strings. These specify the Instance and Class names

Reply:
Build index of desktops it's not compositor task I think.

Reply:
> A compositor that really wants to do the right thing would build an index of desktop entries and convert the class name to an app_id when necessary. Sway won't.

Well, I guess only when **possible** because all applications don't have an desktop entry.

If there was a service or something else which would provide easy lookup of WMClass in desktop entries, would sway be interested in supporting that? (optional support I'll assume)